src/NuGet.Core/NuGet.Protocol/LocalRepositories/FindLocalPackagesResourceV3Provider.cs

```diff
@@ -18,8 +18,10 @@ public FindLocalPackagesResourceV3Provider()
         public override async Task<Tuple<bool, INuGetResource>> TryCreate(SourceRepository source, CancellationToken token)
         {
             FindLocalPackagesResource curResource = null;
+            var feedType = await source.GetFeedType(token);
 
-            if (await source.GetFeedType(token) == FeedType.FileSystemV3)
+            if (feedType == FeedType.FileSystemV3
```

Comment:
What does this do? 

Feels like you are trying to fix 2 problems in 1 PR here.